File: backend/transaction-service/transaction-domain/transaction-application-service/src/main/java/com/saver/system/transaction/service/domain/mapper/TransactionDataMapper.java
package com.saver.system.transaction.service.domain.mapper;

import com.saver.system.domain.valueobject.AccountId;
import com.saver.system.domain.valueobject.Money;
import com.saver.system.domain.valueobject.UserId;
import com.saver.system.transaction.service.domain.dto.create.CreateTransactionCommand;
import com.saver.system.transaction.service.domain.dto.create.CreateTransactionResponse;
import com.saver.system.transaction.service.domain.entity.Transaction;
import com.saver.system.transaction.service.domain.valueobject.TransactionAddress;
import org.springframework.stereotype.Component;

@Component
public class TransactionDataMapper {
    public Transaction createTransactionCommandToOrder(CreateTransactionCommand createTransactionCommand) {
        return Transaction.builder()
                .amount(new Money(createTransactionCommand.getTransactionAmount()))
                .category(createTransactionCommand.getCategory())
                .accountId(new AccountId(createTransactionCommand.getAccountId()))
                .transactionAddress((createTransactionCommand.getAddress()))
                .userId(new UserId(createTransactionCommand.getUserId()))
                .transactionType(createTransactionCommand.getTransactionType())
                .build();
    }
    public CreateTransactionResponse transactionToCreateTransactionResponse(Transaction transaction, String message) {
        return CreateTransactionResponse.builder()
                .transactionAmount(transaction.getMoney())
                .accountId(transaction.getAccountId().getValue())
                .transactionAddress(transaction.getTransactionAddress())
                .userId(transaction.getUserId().getValue())
                .message(message)
                .build();
    }
}

File: old-code/old backend - node/services/transactions/src/domain/domain-application-service/mapper/transaction-data-mapper.ts
import { Money, TransactionId } from "@machsan-tiggo/common";
import { CashTransaction } from "../../domain-core/entity/cash-transaction";
import { Transaction } from "../../domain-core/entity/transaction";
import { CreateTransactionCommand } from "../dto/create/create-transaction-command";
import { CreateTransactionResponse } from "../dto/create/create-transaction-response";

export class TransactionDataMapper {
  public static createCashTransactionCommandToTransaction(
    createTransactionCommand: CreateTransactionCommand
  ): CashTransaction {
    const transaction = new CashTransaction(
      new TransactionId("TODO:ADD ID" + Math.random()),
      createTransactionCommand.userId,
      createTransactionCommand.accountId,
      createTransactionCommand.amount,
      createTransactionCommand.transactionStatus,
      createTransactionCommand.category,
      createTransactionCommand.created_at,
      createTransactionCommand.updated_at,
      createTransactionCommand.transactionAddress
    );

    return transaction;
  }
  public static cashTransactionToCreateTransactionResponse(
    transaction: CashTransaction,
    message: string
  ): CreateTransactionResponse {
    const createTransactionResponse = new CreateTransactionResponse()
      .setUserId(transaction.getUserId().getValue())
      .setAccountId(transaction.getAccountId().getValue())
      .setTransactionAmount(transaction.getAmount())
      .setMessage(message)
      .setTransactionAddress(transaction.getTransactionAddress());

    return createTransactionResponse;
  }
}
